### ISSUE TITLE
takeElementScreenshot takes screenshot of the element correctly but throw uncaught object "[object object]" error right after

### ISSUE THREAD

**Issue by sophiecdd** [DevExpress/testcafe#3189]
In my test I do the following to take a screenshot of an element:
.takeElementScreenshot(Selector('.element'), testScreenshotPath + imgName, '');

my test failed with the following error:
1) **Uncaught object "[object Object]" was thrown**. Throw Error instead.

      Browser: Chrome 70.0.3538 / Windows 10.0.0
      Screenshot: C:\dev\Testcafe-POC\screenshots\2018-12-04_16-32-32\test-3\Chrome_70.0.3538_Windows_10.0.0\errors\1.png

In the screenshot tests folder I can see the screenshot of that element was successfully taken and the content is correct. Can anybody help me with this? Thanks in advance!

**Comment by LavrovArtem** [comment]
This information is not sufficient to find the cause of the issue.
Would you clarify if you reproduced this issue without the `takeScreenshot` action call or with the same steps but without Testcafe.
In addition, try using the [--skip-js-errors](https://devexpress.github.io/testcafe/documentation/using-testcafe/command-line-interface.html#-e---skip-js-errors) flag.
If this does not help, please provide a page or a small example so that we can reproduce it on our side.

**Comment by sophiecdd** [comment]
@andrzej-kodify Thanks for your help! I have tried what you recommended got the error printed out, similar as what you got there, with the try/catch, my test can successfully run through, the screenshot was actually successfully taken in my case, but I do think it is a testcafe issue that at least the error message is supposed to be clear and meaningful right?
{ callsite:
   CallsiteRecord {
     filename: 'C:\\dev\\Testcafe-POC\\ScreenshotTests.js',
     lineNum: 132,
     callsiteFrameIdx: 6,
     stackFrames:
      [ [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        [Object],
        CallSite {},
        [Object],
        [Object],
        [Object],
        [Object],
        CallSite {} ],
     isV8Frames: true },
  isRejectedDriverTask: true }

@LavrovArtem 
With the solution that @andrzej-kodify  posted above, I could get my test run through successfully, since the screenshot of the element was taken correctly. But I wasn't sure how to find out where exactly the error was thrown. In my case, I am trying to take a screenshot of an chart element which located on a canvas, maybe this caused something different than a regular DOM element?

**Comment by andrzej-kodify** [comment]
@LavrovArtem for me it happened with `ClientFunction(() => { return window.location; })` in v0.23.3-alpha2 (while it was working v0.23.2 and before). I've made a workaround to instead return only properties
```ClientFunction(() => {
        const currentLocation = window.location;
        return {
            hash: currentLocation.hash,
            host: currentLocation.host,
            hostname: currentLocation.hostname,
            href: currentLocation.href,
            origin: currentLocation.origin,
            pathname: currentLocation.pathname,
            port: currentLocation.port,
            protocol: currentLocation.protocol,
            search: currentLocation.search,
        };
    }),```
that's the only hint I have, not sure if it'll help to trace the issue

**Comment by miherlosev** [comment]
@sophiecdd @andrzej-kodify

I've tried to run simple test code:
```js
import { ClientFunction } from 'testcafe';
fixture `Fixture`
    .page('https://example.com');
    
test('test', async t => {
    const getLocation = ClientFunction(() => window.location);
    
    console.log(await getLocation());
});
```
and found a problem in our code - https://github.com/DevExpress/testcafe-hammerhead/issues/1866. But, it's not related to the issue you described.
 
The problem is likely related to your website and I need an example showing this issue to find its cause.
Feel free to reopen this issue if you can provide such an example.

**Comment by AndreyBelym** [comment]
This problem is fixed in `testcafe@0.23.3`:
 * https://github.com/DevExpress/testcafe-hammerhead/commit/b3b2cfb3a865950265ae6b77496c242a85442842
* https://github.com/inikulin/replicator/commit/c31578be9d62cb0432593f03e676b4ed3d7586ea

Please reopen the issue if you still can reproduce the problem.
